src/main/resources/assets/react/interfaces/Interfaces.jsx

```diff
@@ -142,27 +163,124 @@ export function Interfaces({
+					collectionIdToDocumentTypeIds[_id] = documentTypeId;
+					collectionIdToFieldKeys[_id] = documentTypeIdToFieldKeys[documentTypeId];
+					collectionIdToName[_id] = _name;
+				});
+				//console.debug('collectionIdToDocumentTypeIds', collectionIdToDocumentTypeIds);
+				//console.debug('collectionIdToFieldKeys', collectionIdToFieldKeys);
+				setCollectionIdToFieldKeys(collectionIdToFieldKeys);
+				//setCollectionIdToDocumentTypeIds(collectionIdToDocumentTypeIds);
 
-		fetch(`${servicesBaseUrl}/interfaceList`)
-			.then(response => response.json())
-			.then(data => {
-				setState((oldState) => {
-					const deref = JSON.parse(JSON.stringify(oldState));
-					//deref.collectionOptions = data.collectionOptions;
-					//deref.collections = data.collections; // Never been used?
-					deref.fieldsObj = data.fieldsObj;
-					deref.interfaces = data.interfaces;
-					deref.stopWordOptions = data.stopWordOptions;
-					deref.synonyms = data.synonyms;
-					deref.thesauriOptions = data.thesauriOptions;
-					return deref;
+				const thesaurusIdToName = {};
+				setThesauriOptions(data.queryThesauri.hits.map(({
+					_id, _name
+				}) => {
+					thesaurusIdToName[_id] = _name;
+					return {
+						key: _id,
+						text: _name,
+						value: _id
+					};
+				}));
+				//console.debug('thesaurusIdToName', thesaurusIdToName);
+				//setThesaurusIdToName(thesaurusIdToName);
+
+				const interfacesObj = {};
+				const interfaceNamesObj = {};
+				data.queryInterfaces.hits.forEach(({
+					_id,
+					_name,
+					collectionIds = [],
+					fields = [], // boost, name
+					stopWords = [],
+					synonymIds = []
+				}) => {
+					interfaceNamesObj[_name] = true;
+					/*const boostableFieldsObj = {
+						'_allText': true // NOTE Hardcode
+					};*/
+					const collectionNamesObj = {};
+					collectionIds.forEach((_id) => {
+						/*const fieldKeys = collectionIdToFieldKeys[_id];
+						if (fieldKeys) {
+							fieldKeys.forEach((fieldKey) => {
+								boostableFieldsObj[fieldKey] = true;
+							});
+						}*/
+						const collectionName = collectionIdToName[_id];
+						if (collectionName) {
+							collectionNamesObj[collectionName] = true;
+						}
+					});
+
+					const thesaurusNames = [];
+					synonymIds.forEach((thesaurusId) => {
+						const thesaurusName = thesaurusIdToName[thesaurusId];
+						if (thesaurusName) {
+							thesaurusNames.push(thesaurusName);
+						}
+					});
+
+					interfacesObj[_id] = {
+						_id,
+						_name,
+						//boostableFieldKeys: Object.keys(boostableFieldsObj).sort(),
+						collectionNames: Object.keys(collectionNamesObj).sort(),
+						fields,
+						stopWords,
+						thesaurusNames
+					};
 				});
-				//setboolIsLoadingService(false);
-			});
+				//console.debug('interfaces', interfaces);
+				setInterfaceNamesObj(interfaceNamesObj);
+				setInterfaces(Object.keys(interfacesObj).map((_id) => interfacesObj[_id]).sort((a,b) => (a._name > b._name) ? 1 : -1));
```

Comment:
This needs to be put on differet lines. 
Its alot of thing happening in a single line of code.